### PR TITLE
Remove using Task.FromResult where possible

### PR DESCRIPTION
**Pull request by rynowak**
Believe it or not, this one in ActionResult is showing in TechEmpower. I
did a pass on all of the product code since the cache task actually
improves the readability of the code :sunglasses:

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Core/ActionResult.cs [hunk 1]
```diff
@@ -22,7 +23,7 @@ public abstract class ActionResult : IActionResult
         public virtual Task ExecuteResultAsync(ActionContext context)
         {
             ExecuteResult(context);
-            return Task.FromResult(true);
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/ActionResult.cs]: 80 bytes is too many

Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/ActionResult.cs]: This is actually around 8% of the allocated bytes in TechEmpower plaintext

**Comment by benaadams** [comment]
:rocket:

**Comment by dougbu** [comment]
:shipit:

**Pull request closed without merging**